reconcile/checkpoint.py

```diff
@@ -0,0 +1,123 @@
+"""Performs an SRE checkpoint.
+
+The checks are defined in
+https://gitlab.cee.redhat.com/app-sre/contract/-/blob/master/content/process/sre_checkpoints.md
+
+"""
+import logging
+import re
+from functools import partial
+from http import HTTPStatus
+from pathlib import Path
+from typing import Any, Iterable, Mapping, Union
+
+import requests
+from jinja2 import Template
+from jira import Issue
+
+from reconcile.utils.constants import PROJ_ROOT
+from reconcile.utils.jira_client import JiraClient
+
+
+DEFAULT_CHECKPOINT_LABELS = ('sre-checkpoint',)
+
+# We reject the full RFC 5322 standard since many clients will choke
+# with some carefully crafted valid addresses. We might
+EMAIL_ADDRESS_REGEXP = re.compile(r'^\w+[-\w\.]*@(?:\w[-\w]*\w\.)+\w+')
+MAX_EMAIL_ADDRESS_LENGTH = 320  # Per RFC3696
+
+MISSING_DATA_TEMPLATE = PROJ_ROOT / 'templates' / \
+    'jira-checkpoint-missinginfo.j2'
+
+
+def url_makes_sense(url: str) -> bool:
+    """Guesses whether the URL may have a meaningful document.
+
+    Obvious cases are if the document can be fully downloaded, but we
+    also accept that the given document may require credentials that
+    we don't have.
```

Comment:
depending on the actual URL field that gets checked here, i wonder if we really are fine with not having access to a document (e.g. sopsURL). i'm not talking about implementing authentication in this context, but rather highlight 401s to the caller of qontract-cli

Reply:
... after some thougth, I don't have a good generic solution in these cases. And right now, the priority is to assist onboardings report quickly on services that list `http://TODO`. Can we leave docs that require authentication to a next iteration?